### DISCUSSION TITLE
Problem Run Blockscout FrontEnd

### DISCUSSION THREAD

**Discussion by ekawibisono** [blockscout/frontend#1118] [Q&A]
Hello, I have installed Blockscout Frontend, but why can't I connect to the backend?
I have done the installation according to the guide but in the developer tool the url reads localhost, even though I have replaced it with the ip address 0.0.0.0

![image](https://github.com/blockscout/frontend/assets/22586872/7acecf72-0b65-4330-83eb-60d574a902ad)

**#First Setup**
`docker run -p 3000:3000 --env-file .env.testnet ghcr.io/blockscout/frontend:latest`

**#Next Step** 
`FRONT_PROXY_PASS=http//host.docker.internal:3000 docker compose -f docker-compose-no-build-external-frontend.yml up -d`

**#This ENV in FrontEND**

```
NEXT_PUBLIC_API_HOST=ip address VPS
NEXT_PUBLIC_API_PROTOCOL=http
NEXT_PUBLIC_STATS_API_HOST=http://ip address VPS:8080/
NEXT_PUBLIC_VISUALIZE_API_HOST=http://ip address VPS:8081/

# Blockchain Configuration
NEXT_PUBLIC_NETWORK_NAME=TestPoS
NEXT_PUBLIC_NETWORK_SHORT_NAME=PoS
NEXT_PUBLIC_NETWORK_RPC_URL=http://ip address VPS:9545/
NEXT_PUBLIC_NETWORK_ID=12120
NEXT_PUBLIC_NETWORK_CURRENCY_NAME=TestPoS
NEXT_PUBLIC_NETWORK_CURRENCY_SYMBOL=PoS
NEXT_PUBLIC_NETWORK_CURRENCY_DECIMALS=18
NEXT_PUBLIC_NETWORK_VERIFICATION_TYPE=validation
NEXT_PUBLIC_HIDE_INDEXING_ALERT=true
NEXT_PUBLIC_IS_TESTNET=true

NEXT_PUBLIC_API_BASE_PATH=0.0.0.0/
NEXT_PUBLIC_APP_HOST=ip address VPS
NEXT_PUBLIC_APP_PROTOCOL=http
# NEXT_PUBLIC_API_WEBSOCKET_PROTOCOL=ws

NEXT_PUBLIC_HOMEPAGE_CHARTS=['daily_txs']
NEXT_PUBLIC_HOMEPAGE_PLATE_TEXT_COLOR=rgb(149, 196, 84)
NEXT_PUBLIC_HOMEPAGE_PLATE_BACKGROUND=radial-gradient(103.03% 103.03% at 0% 0%, rgb(49, 44, 44) 0%, rgb(49, 44, 44) 100%)
NEXT_PUBLIC_HOMEPAGE_SHOW_GAS_TRACKER=true
NEXT_PUBLIC_HOMEPAGE_SHOW_AVG_BLOCK_TIME=true
```

Please Help to Fix, and Discuss about this problem

**Comment by ekawibisono** [comment]
Hello, I Found error about using URL in screenshot

![image](https://github.com/blockscout/frontend/assets/22586872/095f4bd6-684e-4f54-81a6-e1f7f505c598)

When I use docker-compose-no-build-frontend.yml it shows error like in the screenshot, same as when I run using docker-compose-no-build-external-frontend.yml

**Comment by ekawibisono** [comment]
Now Found error in Proxy, using docker compose -f docker-compose-no-build-external-frontend.yml up -d,

![image](https://github.com/blockscout/frontend/assets/22586872/9913a797-ed31-4185-8d82-0f9b91622b30)

**Reply by vbaranov** [reply]
@ekawibisono when you run `docker-compose-no-build-external-frontend.yml` config you should provide URL to the external frontend. PLease take a look https://docs.blockscout.com/for-developers/frontend-migration/separate-frontend#4-run-docker-compose-with-front_proxy_pass-variable.

**Reply by ekawibisono** [reply]
@vbaranov Running it like this for the frontend ?
`FRONT_PROXY_PASS=http//ipaddressvps:3000 docker compose -f docker-compose-no-build-external-frontend.yml up`

**Reply by vbaranov** [reply]
Yes

**Reply by ekawibisono** [reply]
@vbaranov I want to ask, why is port 8050 when I check in the terminal not working? for visualizer and stats

**Comment by ekawibisono** [comment]
Sorry beforehand, apparently it's been solved, my browser is Microsoft Edge Blocking CORS, and after I installed the CORS extension it can be read. however there is still a problem on the Proxy section when building using docker compose -f docker-compose-no-build-external-frontend.yml

**Comment by tom2drum** [comment]
@vbaranov can you help here with the proxy setup?

from my point of view this variable is incorrect - `NEXT_PUBLIC_API_BASE_PATH=0.0.0.0/`
also please check that you are passing into `NEXT_PUBLIC_API_HOST` only hostname without protocol and port

**Reply by ekawibisono** [reply]
OK, I'll do that first

**Comment by ekawibisono** [comment]
Hello, I want to ask, why is port 8050 not being read on my port? I've set the firewall and it's still not readable. is there any solution ?

cc : @vbaranov @tom2drum 

![image](https://github.com/blockscout/frontend/assets/22586872/5bef3596-2b9c-496f-905b-d488fa3f4cee)

**Comment by ekawibisono** [comment]
Hello i have a problem with proxy, can you help me to this topic ?
https://github.com/blockscout/blockscout/issues/8388

CC : @vbaranov @tom2drum
